src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable */
+import classNames from "classnames";
+
+import React, { useContext, useEffect, useState } from "react";
+import { Todo } from "../../types/Todo";
+import { TodosContext } from "../../context/TodosContext";
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { todosLoading, onDeleteTodo, toggleTodoStatus, updateTodo } =
+    useContext(TodosContext);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [updateField, setUpdateField] = useState(todo.title || "");
+  const [firtTitle, setFirstTitle] = useState(todo.title);
+
+  useEffect(() => {
+    setFirstTitle(todo.title || "");
+    setUpdateField(todo.title || "");
+  }, [todo]);
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      setIsEditing(false);
+    }
+
+    if (event.key === "Escape") {
+      setIsEditing(false);
+      setUpdateField(firtTitle);
+
+      return;
+    }
+  };
+
+  const handleSubmit = (event?: React.FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+
+    if (!updateField) {
+      onDeleteTodo(todo.id);
+
+      return;
+    }
+
+    updateTodo(updateField, todo.id);
+  };
+
+  return (
+    <div className={classNames({ todo: true, completed: todo.completed })}>
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={() => toggleTodoStatus(todo.id)}
+        />
+      </label>
+
+      {isEditing ? (
+        <form onSubmit={handleSubmit}>
+          <input
+            value={updateField}
+            onChange={(event) => setUpdateField(event.target.value)}
+            type="text"
+            className="todo__title-field"
+            onKeyUp={handleKeyUp}
+            onBlur={() => handleSubmit}
```

Comment:
1. If on click outside input blur styles are not removed and input title not changed
2. fix the ability to add only spaces to input title
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/54936044/c607798c-e710-4068-9c62-ee63b3d30b7c)